boot/boot.js

```diff
@@ -287,6 +287,92 @@ $tw.utils.decodeURIComponentSafe = function(s) {
 	return v;
 };
 
+/*
+Helpers for encoding/decoding URIs
+*/
+
+/*
+The character that will substitute for a space in the URL
+*/
+var SPACE = "_";
+/*
+The character that will separate out the list elements in the URL
+*/
+var CONJUNCTION = ";";
+/*
+Those of the allowed url characters claimed by TW
+*/
+var claimed= [SPACE, ":", CONJUNCTION];
+/*
+Non-alphanumeric characters allowed in a URL fragment
+More information at https://www.rfc-editor.org/rfc/rfc3986#appendix-A
+*/
+var pchar = "-._~!$&'()*+,;=:@/?";
```

Comment:
I think `CLAIMED` and `PCHAR` could be pseudo-constants too. We use "pseudo constants" quite a bit. So IMO it would be consistent with the rest of the core code

Reply:
Sure.  I don't know if PCHAR (regardless of capitalization) is too technical.  It's meaningful only in context of the spec.  I just didn't find a good alternative name.

Reply:
You could use 

```
var VALID_IN_URL="-._~!$&'()*+,;=:@/?".split();
```

Since `claimed` is an array. May be VALID_IN_URL should be an array too. It would save you the `.split()` later in the code. 
But that's only nit-picking.

Reply:
I did switch to a variant of this name in the latest commit, but won't bother making it an array, as this should be thrown away after the bikeshedding on characters to use is done.